Kinvey/Kinvey/User.swift

```diff
@@ -1297,6 +1300,35 @@ open class User: NSObject, Credential, Mappable {
             completionHandler: completionHandler
         )
     }
+    
+    /**
+     Login with MIC using Automated Authorization Grant Flow. We strongly recommend use [Authorization Code Grant Flow](http://devcenter.kinvey.com/rest/guides/mobile-identity-connect#authorization-grant) instead of [Automated Authorization Grant Flow](http://devcenter.kinvey.com/rest/guides/mobile-identity-connect#automated-authorization-grant) for security reasons.
+     */
+    @discardableResult
+    open class func login<U: User>(
+        username: String,
+        password: String,
+        provider: AuthProvider,
+        options: Options? = nil,
+        completionHandler: ((Result<U, Swift.Error>) -> Void)? = nil
+    ) -> AnyRequest<Result<U, Swift.Error>> {
+        switch provider {
+        case .kinvey:
```

Comment:
Is this the default `AuthProvider` if none is provided?

Reply:
no, actually this was the best way that I find to allow overload the `login()` method without need to name it like `loginMIC()`, which in Swift seems weird. So the user actually have to specify `.mic` or `.kinvey` otherwise the compiler would not know if you are trying to login with your Kinvey username/password or with your MIC username/password

Reply:
I think this is a great solution to prevent having `loginMIC()`. However, it looks like that a user will have to specify a `provider` which will break existing apps. Can't we just specify `AuthProvider.kinvey` as the default? Something like this:

```swift
open class func login<U: User> (
  username: String,
  password: String,
  provider: AuthProvider? = .kinvey,
  completionHandler: ((Result<U, Swift.Error>) -> Void)? = nil
) -> AnyRequest<Result<U, Swift.Error>> {
 //...
}
```

_Not sure my syntax is correct but something like that_

Reply:
I tried that too, but the problem is if you specify a default value there's no way for the compiler to know which `login()` overload are you referring to. We are also deprecating the old way, so no breaking changes. Let me try if I can eliminate the previous kinvey `login()` method and replace for the new method with the `.kinvey` as default value and see if it will not cause any breaking change. going to test this now.

Reply:
@thomasconner i was able to move the previous implementation to be inside the new method and set the default value, so no breaking changes and also keep a cleaner code. thanks for your suggestion!

Reply:
@thomasconner please take another look, it should be good now. thanks for your suggestion!